runtime/databricks/automl_runtime/forecast/pmdarima/training.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2021 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import List
+
+import pandas as pd
+
+from databricks.automl_runtime.forecast.pmdarima.diagnostics import generate_cutoffs, cross_validation
+from databricks.automl_runtime.forecast import OFFSET_ALIAS_MAP
+
+
+class ArimaEstimator:
+    """
+    ARIMA estimator using pmdarima.auto_arima.
+    """
+
+    def __init__(self, horizon: int, frequency_unit: str, metric: str, seasonal_periods: List[int],
+                 num_folds: int = 5, max_eval: int = 10) -> None:
+        """
+        :param horizon: Number of periods to forecast forward
+        :param frequency_unit: Frequency of the time series
+        :param metric: Metric that will be optimized across trials
+        :param seasonal_periods: A list of seasonal periods for tuning.
+        :param num_folds: Number of folds for cross validation
+        :param max_eval: Max steps
+        """
+        self._horizon = horizon
+        self._frequency_unit = frequency_unit
+        self._metric = metric
+        self._seasonal_periods = seasonal_periods
+        self._num_folds = num_folds
+
+    def fit(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Fit the ARIMA model with tuning of seasonal period m and with pmdarima.auto_arima.
+        :param df: A pd.DataFrame containing the history data. Must have columns ds and y.
+        :return: A pd.DataFrame with the best model (pickled) and its metrics from cross validation.
+        """
+        import pandas as pd
+        import pickle
```

Comment:
We don't need to import these in the function, right?

Reply:
I felt I did this previously to make the multiseries forecasitng work. The prophet estimator is also doing something similar (https://github.com/databricks/automl/blob/468206f3161546beeeb5bff1cc1c46d69accc41d/runtime/databricks/automl_runtime/forecast/prophet/forecast.py#L124). Let me double-check though.

Reply:
It doesn't have to be inside the method. Checked and fixed.